File: chain-of-responsibility/python/jpeg_handler.py
# Created by Sezer BOZKIR<admin@sezerbozkir.com> at 20.12.2021
from abstract_handler import AbstractHandler
from image import Image


class JPEGHandler(AbstractHandler):
    """
    JPEG dosyalarını dönüştüren sınıf.
    UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
    """

    def handle_request(self, image: Image):
        if image.extension == "JPEG":
            # JPEG işlemine ait dönüştürme kodları işlemleri.
            print("JPEG to PNG")
        else:
            print("not related to JPEG Handler.")
            # Bu sınıfa ait bir işlem değilse zincirin bir sonraki halkasına aktarılır.
            return super().handle_request(image)

File: chain-of-responsibility/go/jpeg-handler.go
package main

import "fmt"

// JPEG dosyalarını dönüştüren sınıf.
// UML diyagramındaki ConcreteHandler sınıfına denk gelmektedir.
type jpegHandler struct {
	nextHandler BaseHandler
}

func NewJPEGHandler() *jpegHandler {
	return &jpegHandler{}
}

func (h *jpegHandler) SetNextHandler(nextHandler BaseHandler) {
	h.nextHandler = nextHandler
}

func (h jpegHandler) HandleRequest(img Image) {
	if img.Extension == "JPEG" {
		fmt.Printf("JPEG to PNG")
		// JPEG işlemine ait dönüştürme kodları işlemleri.
	} else {
		// Bu sınıfa ait bir işlem değilse zincirin bir sonraki halkasına aktarılır.
		h.nextHandler.HandleRequest(img)
	}
}
